src/main/java/com/teragrep/rlp_03/SyslogFrameProcessor.java

```diff
@@ -65,70 +67,34 @@ public class SyslogFrameProcessor implements FrameProcessor, AutoCloseable {
+        this.relpCommandConsumerMap.put(RelpCommand.OPEN, new RelpEventOpen());
+        this.relpCommandConsumerMap.put(RelpCommand.SYSLOG, new RelpEventSyslog());
+
+        this.relpEventServerClose = new RelpEventServerClose();
     }
 
     @Override
     public void accept(FrameContext frameContext) {
         // TODO add TxID checker that they increase monotonically
 
-        List<RelpFrameTX> txFrameList = new ArrayList<>(); // FIXME
-        switch (frameContext.relpFrame().command().toString()) {
-            case RelpCommand.ABORT:
-                // abort sends always serverclose
-                txFrameList.add(createResponse(frameContext.relpFrame(), RelpCommand.SERVER_CLOSE, ""));
-                break;
-
-            case RelpCommand.CLOSE:
-                // close is responded with rsp
-                txFrameList.add(createResponse(frameContext.relpFrame(), RelpCommand.RESPONSE, ""));
-
-
-                // closure is immediate!
-                txFrameList.add(createResponse(frameContext.relpFrame(), RelpCommand.SERVER_CLOSE, ""));
-                break;
-
-            case RelpCommand.OPEN:
-                String responseData = "200 OK\nrelp_version=0\n"
-                        + "relp_software=RLP-01,1.0.1,https://teragrep.com\n"
-                        + "commands=" + RelpCommand.SYSLOG + "\n";
-                txFrameList.add(createResponse(frameContext.relpFrame(), RelpCommand.RESPONSE, responseData));
-                break;
-
-            case RelpCommand.RESPONSE:
-                // client must not respond
-                txFrameList.add(createResponse(frameContext.relpFrame(), RelpCommand.SERVER_CLOSE, ""));
-                break;
-
-            case RelpCommand.SERVER_CLOSE:
-                // client must not send serverclose
-                txFrameList.add(createResponse(frameContext.relpFrame(), RelpCommand.SERVER_CLOSE, ""));
-                break;
-
-            case RelpCommand.SYSLOG:
-                if (frameContext.relpFrame().payload().size() > 0) {
-                    try {
-                        cbFunction.accept(frameContext);
-                        txFrameList.add(createResponse(frameContext.relpFrame(), RelpCommand.RESPONSE, "200 OK"));
-                    } catch (Exception e) {
-                        LOGGER.error("EXCEPTION WHILE PROCESSING SYSLOG PAYLOAD", e);
-                        txFrameList.add(createResponse(frameContext.relpFrame(),
-                                RelpCommand.RESPONSE, "500 EXCEPTION WHILE PROCESSING SYSLOG PAYLOAD"));
-                    }
-                } else {
-                    txFrameList.add(createResponse(frameContext.relpFrame(), RelpCommand.RESPONSE, "500 NO PAYLOAD"));
-
-                }
-                break;
+       String relpCommand = frameContext.relpFrame().command().toString();
 
-            default:
-                break;
+       Consumer<FrameContext> commandConsumer = relpCommandConsumerMap.get(relpCommand);
```

Comment:
Using `relpCommandConsumerMap.getOrDefault(relpCommand, new RelpEventServerClose())` should be fine?